### ISSUE TITLE
[stable/nginx-ingress] Helm install nginx-ingress with parameter controller.nodeSelector does not work

### ISSUE THREAD

**Issue by wwyhy** [helm/charts#12961]
**Version of Helm and Kubernetes**:
Kubernetes: v1.14.0
Helm: v2.13.1

**Which chart**:
stable/nginx-ingress 

**What happened**:
# helm install stable/nginx-ingress \
  --namespace ingress-nginx \
  --name nginx-ingress \
  --set controller.hostNetwork=true \
  --set controller.kind=DaemonSet \
  --set controller.service.type="" \
  --set controller.daemonset.useHostPort=true \
  --set controller.daemonset.hostPorts.http=80 \
  --set controller.daemonset.hostPorts.https=443 \
  --set controller.nodeSelector.master=true

I have labeled the 3 master with "master=true", I got the error: 
Error: release nginx-ingress failed: DaemonSet in version "v1beta1" cannot be handled as a DaemonSet: v1beta1.DaemonSet.Spec: v1beta1.DaemonSetSpec.Template: v1.PodTemplateSpec.Spec: v1.PodSpec.NodeSelector: ReadString: expects " or n, but found t, error found in #10 byte of ...|"master":true},"serv|..., bigger context ...|rst","hostNetwork":true,"nodeSelector":{"master":true},"serviceAccountName":"nginx-ingress","termina|...

**What you expected to happen**:
The nginx-ingress-controller should be assigned the 3 master which labeled "master=true".
but it seems that the parameter "--set controller.nodeSelector" does not work.

**How to reproduce it** (as minimally and precisely as possible):
1. label 3 master with  "master=true"
2. run install
# helm install stable/nginx-ingress \
  --namespace ingress-nginx \
  --name nginx-ingress \
  --set controller.hostNetwork=true \
  --set controller.kind=DaemonSet \
  --set controller.service.type="" \
  --set controller.daemonset.useHostPort=true \
  --set controller.daemonset.hostPorts.http=80 \
  --set controller.daemonset.hostPorts.https=443 \
  --set controller.nodeSelector.master=true

**Anything else we need to know**:
In this case, i expect the nginx-ingress-controller only running on the 3 master, but nodes.
so I used the parameter " --set controller.nodeSelector "

**Comment by ChiefAlexander** [comment]
This should be resolved with #13646.
Follow the instructions on the Readme for doing the upgrade: https://github.com/helm/charts/tree/master/stable/nginx-ingress#helm-error-when-upgrading-specclusterip-invalid-value-

**Comment by ChiefAlexander** [comment]
The issue here is that `true` is getting evaluated as a boolean instead of what kubernetes expects (a string). This is a limitation of the `--set` flag: https://helm.sh/docs/using_helm/#the-format-and-limitations-of-set
> Helm will cast certain values specified with --set to integers. For example, --set foo=true results Helm to cast true into an int64 value. In case you want a string, use a --set’s variant named --set-string. --set-string foo=true results in a string value of "true".

Using set-string this will work as intended. However while looking into this issue I found that we are not wrapping the annotations field of the daemonset which will produce an error if it is not set. I will create another issue to fix that up. For now you will have to pass in:
```
--namespace ingress-nginx
--name nginx-ingress
--set controller.hostNetwork=true
--set controller.kind=DaemonSet
--set controller.service.type=""
--set controller.daemonset.useHostPort=true
--set controller.daemonset.hostPorts.http=80
--set controller.daemonset.hostPorts.https=443
--set-string controller.nodeSelector.master=true
--set controller.podAnnotations.something=something
```
